sklearn/datasets/_samples_generator.py

```diff
@@ -1241,6 +1243,18 @@ def make_low_rank_matrix(
 
 # TODO(1.3): Change argument `data_transposed` default from True to False.
 # TODO(1.3): Deprecate data_transposed, always return data not transposed.
+
+
+@validate_params(
+    {
+        "n_samples": [Interval(Integral, 0, None, closed="neither")],
+        "n_components": [Interval(Integral, 0, None, closed="neither")],
+        "n_features": [Interval(Integral, 0, None, closed="neither")],
+        "n_nonzero_coefs": [Interval(Integral, 0, None, closed="neither")],
+        "random_state": ["random_state"],
+        "data_transposed": ["boolean"],
```

Comment:
We raise a warning using the default values. We can use the following to take into account the `"warn"`.

```suggestion
        "data_transposed": ["boolean", Hidden(StrOptions({"warn"}))],
```

Reply:
fixed it!